DisjointSet.java

```diff
@@ -0,0 +1,47 @@
+
+public class DisjointSet {
```

Comment:
Adding comments as to how the set operates would make it easier to read here. None of the public methods have any explanation.